File: infrastruktur/src/test/java/com/github/haschi/haushaltsbuch/infrastruktur/SystemtestPicoFactory.java
package com.github.haschi.haushaltsbuch.infrastruktur;

import cucumber.runtime.java.picocontainer.PicoFactory;

public class SystemtestPicoFactory extends PicoFactory
{
    public SystemtestPicoFactory()
    {
        addClass(Testcloud.class);
        addClass(Welt.class);
    }
}

File: infrastruktur/src/test/kotlin/com/github/haschi/haushaltsbuch/infrastruktur/SystemtestPicoFactory.kt
package com.github.haschi.haushaltsbuch.infrastruktur

import com.github.haschi.dominium.haushaltsbuch.core.application.Anwendungskonfiguration
import cucumber.runtime.java.picocontainer.PicoFactory
import io.vertx.core.logging.Logger
import io.vertx.core.logging.LoggerFactory

class SystemtestPicoFactory : PicoFactory()
{
    init
    {
        logger.info("Initialisiere Systemtest Pico Factory")
        addClass(Testcloud::class.java)
        addClass(Welt::class.java)
        addClass(AxonInfrastrukturFactory::class.java)
        addClass(Anwendungskonfiguration::class.java)
    }

    companion object
    {
        val logger: Logger = LoggerFactory.getLogger(SystemtestPicoFactory::class.java)
    }
}
